unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/QuickBarController.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public interface IQuickBarController
+{
+    event Action<int> OnQuickBarShortcutSelected;
+    event Action<CatalogItem> OnCatalogItemSelected;
+
+    void Initialize(IQuickBarView view, ISceneCatalogController sceneCatalogController);
+    void Dispose();
+    int GetSlotsCount();
+    CatalogItem QuickBarObjectSelected(int index);
+    void SetIndexToDrop(int index);
+    void SceneObjectDropped(BaseEventData data);
+    void QuickBarInput(int quickBarSlot);
+}
+
+public class QuickBarController : IQuickBarController
+{
+    public event Action<int> OnQuickBarShortcutSelected;
+    public event Action<CatalogItem> OnCatalogItemSelected;
+
+    internal IQuickBarView quickBarView;
+    internal ISceneCatalogController sceneCatalogController;
+
+    internal CatalogItem[] quickBarShortcutsCatalogItems = new CatalogItem[AMOUNT_OF_QUICK_SLOTS];
+    internal int lastIndexDroped = -1;
+
+    const int AMOUNT_OF_QUICK_SLOTS = 9;
```

Comment:
can we move the constant to the very top of the class?

Reply:
Of course :) Done in https://github.com/decentraland/explorer/pull/2103/commits/66ed74b073b7d18de5a29f3fbcace90ddfcf293c